### PR TITLE
Collapsify ❤️ Reactivity

### PR DESCRIPTION
**Pull request by terinjokes**
CP-48 requires better communication between the Collapsify service and our internal processes to be able to give better error messages to users. The crux of this branch rests upon this requirement.

Instead of utilizing Promises to their full potential, the base branch would very quickly "catch" the failing Promises returned by the HTTP library and return a resolved Promise containing an empty Buffer. The rest of the service was centered on this behavior. :person_frowning:

This pull request refactors Collapsify onto the reactive model proposed by Reactive Extensions, and RxJS, and ensures standardized errors are returned by the Collapsify service in such a way that a client can understand the cause of the error.

### CODE REVIEW

File: lib/utils/httpclient.js [hunk 1]
```diff
@@ -1,36 +1,74 @@
...
+        connection: 'keep-alive',
+        user_agent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10; rv:33.0) Gecko/20100101 Firefox/33.0 Collapsify/' + VERSION + ' node/' + process.version,
+        agent: /^https/.test(currentURL) ? httpsAgent : httpAgent
+      }, function(err, res) {
+        if (err) {
+          return observer.onError(err);
+        }
+
+        if (res.statusCode >= 400) {
+          return observer.onError(new (errors.find(res.statusCode))('HTTP ' + res.statusCode + ': ' + url));
```
Comment (sebmck) [inline lib/utils/httpclient.js]: `url` should be `currentURL` or else it'll reference the import.

**Comment by dotjs** [comment]
Tests, tests and more test required.

**Comment by dotjs** [comment]
Implement observable for startkeyframerule event.

**Comment by terinjokes** [comment]
@dotjs woot! this last commit finally splits the flatteners logic into separate files.

**Pull request closed without merging**